src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -45,11 +45,14 @@ public JsonConverter GetConverter(Type typeToConvert)
                 ThrowHelper.ThrowArgumentNullException(nameof(typeToConvert));
             }
 
-            if (JsonSerializer.IsReflectionEnabledByDefault && _typeInfoResolver is null)
+            if (JsonSerializer.IsReflectionEnabledByDefault)
```

Comment:
Unrelated to the other changes, but it appears that the previous method structure resulted in rooted reflection when the feature switch is disabled in the trimming tests. Is this a known issue of the linker?

cc @vitek-karas @eerhardt